human-readable-types/src/main/java/com/palantir/humanreadabletypes/HumanReadableByteCount.java

```diff
@@ -172,16 +173,55 @@ private HumanReadableByteCount(long size, ByteUnit unit) {
         this.unit = Preconditions.checkNotNull(unit, "unit must not be null");
     }
 
+    /**
+     * Reads the value of this {@link HumanReadableByteCount} as a tuple of quantity and unit.
+     * Unlike {@link #getSize()}, this API is meant to make it obvious that the data is only meaningful
+     * when both pieces are used together.
+     *
+     * @param <T> Result type.
+     */
+    @FunctionalInterface
+    public interface SizeFunction<T> {
+
+        /**
+         * A method allowing the raw quantity and unit values to be interpreted into a more specific type.
+         *
+         * @param quantity Raw number of the given {@code unit}
+         * @param unit Unit which the {@code quantity represents}
+         * @return A representation of the byte-count value
+         */
+        T apply(long quantity, ByteUnit unit);
+    }
+
+    /**
+     * Provides the value of this {@link HumanReadableByteCount} as a tuple of quantity and unit to the
+     * {@link SizeFunction}.
+     *
+     * @param <T> Result type.
+     */
+    public <T> T map(SizeFunction<T> function) {
+        Preconditions.checkNotNull(function, "function must not be null");
+        return Preconditions.checkNotNull(function.apply(size, unit), "result must not be null");
+    }
+
     /**
      * The size of this byte string in {@link HumanReadableByteCount.ByteUnit binary units}.
+     *
+     * @deprecated in favor of {@link #map(SizeFunction)} which is harder to misuse.
      */
+    @DoNotCall
```

Comment:
Will this fail the build for people? Do we need to write a migration for them?

Reply:
Yes, but it won’t cause transitive failures. The problem with a migration is that it won’t fix incorrect uses.